### PR TITLE
592 - Integrate content picker with piping menu

### PR DESCRIPTION
**Pull request by tgandrews**
### What is the context of this PR?
Integrates the ContentPicker with the new piping menu.

### How to review 
1. Ensure you can add piped values from previous answers.
2. Ensure the tests pass.

### Dependencies
- [x] - Content Picker component - #424

### CODE REVIEW

**Comment by tgandrews** [comment]
[Hooking content picker into the existing piping ui.](https://trello.com/c/PQWO7GWS/615-hooking-content-picker-into-the-existing-piping-ui)

**Comment by adammahmood** [comment]
Yet to review code but a few minor things I have noticed when testing the feature locally:

1) Section/Question/Answer is marked as "Unlabelled Option" even though I have a Title/Label.  Should this not be using `displayName` and utilise the title or fallback to our default?
2) When opening the content picker, you can click on the "Section" accordion to open a list of Sections, however, you can't close the list until you select an item.  You can only click on an item.  Is this intended?  
3) When deleting a piped answer, the updates aren't reflected on the input where the value is being piped. A hard reload fixes this. There's an [existing issue](https://github.com/ONSdigital/eq-author/issues/428) where the sidebar isn't updated, however, the input was still updating.

**Comment by tgandrews** [comment]
1 and 3. I presume these are related. I will look into it.
2. Yes, this as designed.

**Comment by adammahmood** [comment]
@tgandrews Looks like 1 and 3 are now resolved. 👍

File: src/components/RichTextEditor/PipingMenu.test.js [hunk 1]
```diff
@@ -81,7 +81,7 @@ describe("PipingMenu", () => {
     match = createMatch(questionnaire.id, section.id, page.id);
     const wrapper = render();
 
-    expect(wrapper.prop("disabled")).toBe(true);
+    expect(wrapper.find("PipingMenu__MenuButton").prop("disabled")).toBe(true);
```
Comment (adammahmood) [inline src/components/RichTextEditor/PipingMenu.test.js]: Is it better to export `MenuButton` and use `expect(wrapper.find(MenuButton).prop("disabled")).toBe(true);` ?

**Comment by adammahmood** [comment]
A couple of things relating to the content picker (which I appreciate isn't part of this PR) and I can't comment on the lines:

1) `src/components/ContentPicker/ContentPickerSingle.js`  
`data: PropTypes.any // eslint-disable-line react/forbid-prop-types`.  Why do we allow any structure of `data` when we expect the object to be a certain shape when it's used:

```
  {data.map(option => {
          return (
            <PickerOption
              id={`option-${option.id}`}
              key={option.id}
              selected={selectedOption === option.id}
              onClick={() => onOptionClick(option)}
            >
              {option.displayName}
            </PickerOption>
          );
        })}
```

2) Same question for `src/components/ContentPickerModal`

File: cypress/integration/authenticated/piping_spec.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    cy.get(testId("nav-page-link"))
+      .last()
+      .click();
+
+    cy.get(testId("txt-question-title", "testid")).focus();
+    cy.get(testId("piping-button"))
+      .first()
+      .click();
+
+    cy.get("#Section-title").click();
```
Comment (adammahmood) [inline cypress/integration/authenticated/piping_spec.js]: Why is the component capitalising these ids?  We usually keep all ids lower case.

Comment (tgandrews) [reply cypress/integration/authenticated/piping_spec.js]: It's deriving them from input but can change.

**Comment by tgandrews** [comment]
@adammahmood I have updated to remove the eslint ignores and specify what we actually require. I have also changed to remove the ids which were using title which was wrong as this was a text to display to the user.

**Comment by samiwel** [comment]
image
Nice 👍

**Pull request merged**